ts/features/idpay/initiative/details/saga/__test__/handleGetInitiativeDetails.test.ts

```diff
@@ -0,0 +1,96 @@
+import * as E from "fp-ts/lib/Either";
+import { expectSaga } from "redux-saga-test-plan";
+import { PreferredLanguageEnum } from "../../../../../../../definitions/backend/PreferredLanguage";
+import {
+  InitiativeDTO,
+  StatusEnum
+} from "../../../../../../../definitions/idpay/wallet/InitiativeDTO";
+import { appReducer } from "../../../../../../store/reducers";
+import { NetworkError } from "../../../../../../utils/errors";
+import { idpayInitiativeGet } from "../../store";
+import { handleGetInitiativeDetails } from "../handleGetInitiativeDetails";
+
+const mockResponseSuccess: InitiativeDTO = {
+  initiativeId: "123",
+  status: StatusEnum.REFUNDABLE,
+  endDate: new Date(),
+  nInstr: 123
+};
+
+const mockFailure: NetworkError = {
+  kind: "generic",
+  value: new Error("response status code 401")
+};
+const mockToken = "mock";
+const mockLanguage = PreferredLanguageEnum.it_IT;
+
+describe("Test IDPay initiative details saga", () => {
+  it("should call the success handler on success", async () => {
+    const getWallet = jest.fn();
+    getWallet.mockImplementation(() =>
+      E.right({ status: 200, value: mockResponseSuccess })
+    );
+
+    await expectSaga(
+      handleGetInitiativeDetails,
+      getWallet,
+      mockToken,
+      mockLanguage,
+      { initiativeId: "123" }
+    )
+      .withReducer(appReducer)
+      .put(idpayInitiativeGet.success(mockResponseSuccess))
+      .run();
+  });
+  it("should call the failure handler on failure", async () => {
+    const getWallet = jest.fn();
+    getWallet.mockImplementation(() => E.left(mockFailure));
+
+    await expectSaga(
+      handleGetInitiativeDetails,
+      getWallet,
+      mockToken,
+      mockLanguage,
+      { initiativeId: "123" }
+    )
+      .withReducer(appReducer)
+      .put(idpayInitiativeGet.failure(mockFailure))
+      .run();
+  });
```

Comment:
Left is called only when there is an API exception. An API error (status code != 200) is usually a Right with an ErrorDTO object inside.